### PR TITLE
dante: disable getaddrinfo() checks if cross

### PR DESCRIPTION
**Pull request by ghost**
###### Description of changes

This PR contains three changes to `dante` which would merge-conflict with each other if submitted separately.

The first commit fixes breakage resulting from a recent upgrade to autoconf:

- dante: include gnugrep and gawk in nativeBuildInputs if autoreconfHook-ing

The second commit runs `autoreconfHook` unconditionally, rather than only on one platform, so we notice problems more quickly:

- dante: autoreconfHook unconditionally

The third change expands the set of conditions under which we bypass dante's configure-time `getaddrinfo()` check to include not only mips native builds, but also all cross builds.  The bypass is required in both cases:

- dante: disable getaddrinfo() checks if cross

### CODE REVIEW

**Comment by amjoseph-nixpkgs** [comment]
@ofborg build pkgsCross.mips64el-linux-gnuabi64.dante
@ofborg build pkgsCross.aarch64-multiplatform.dante

**Comment by amjoseph-nixpkgs** [comment]
[pkgsCross.aarch64-multiplatform.dante on x86_64-linux — Success](https://github.com/NixOS/nixpkgs/pull/224929/checks?check_run_id=12556256838)

File: pkgs/servers/dante/default.nix [hunk 1]
```diff
@@ -10,7 +15,7 @@ stdenv.mkDerivation rec {
     sha256 = "0pbahkj43rx7rmv2x40mf5p3g3x9d6i2sz7pzglarf54w5ghd2j1";
   };
 
-  nativeBuildInputs = lib.optional stdenv.hostPlatform.isMips64 autoreconfHook;
+  nativeBuildInputs = [ autoreconfHook gnugrep gawk ];
```
Comment (Artturin) [inline pkgs/servers/dante/default.nix:18]: Weird this shouldn't be necessary
https://github.com/NixOS/nixpkgs/blob/8b2521bdae89126497fa6f139fb9d030d5d13184/pkgs/stdenv/generic/common-path.nix#L6-L7

Comment (amjoseph-nixpkgs) [reply pkgs/servers/dante/default.nix]: Yep, you're right.   74dc0681466c74aab517b706c2dfa10ec54d2e13

**Review by Artturin — APPROVED** [review]

**Pull request merged**